src/delta.c

```diff
@@ -169,14 +170,20 @@ void apply_deltas(struct manifest *current_manifest)
 
 			if (!compute_hash_from_file(filename, hash) || !hash_equal(file->hash, hash)) {
 				free_string(&filename);
+				bad_on_sys = true;
 				continue;
 			}
 
 			found = filename;
 		}
 
 		if (!found) {
-			fprintf(stderr, "Couldn't use delta file %s: no 'from' file to apply was found\n", delta_file);
+			fprintf(stderr, "Couldn't use delta file %s: ", delta_file);
+			if (bad_on_sys) {
+				fprintf(stderr, "'from' file corrupted on system, consider running 'swupd verify --fix'\n");
+			} else {
+				fprintf(stderr, "no 'from' file to apply was found\n");
```

Comment:
Hmm, does this error message now hint at a problem with mixer delta pack creation? That is, if a delta exists in STATEDIR/delta, surely the "from" file is supposed to be installed on the user's system? Because a delta pack wouldn't be downloaded unless the associated bundle was already installed to begin with...

Reply:
It could, but it could also indicate corruption of the user's system. I don't think users will see this specific error much after this commit is released. I think it is much more likely they will see `'from' file corrupted on system` unless they go around randomly removing files.

Reply:
Right, I hope they will see the other error instead :-)